plugins/soundsourcemediafoundation/audiosourcemediafoundation.h

```diff
@@ -0,0 +1,63 @@
+#ifndef AUDIOSOURCEMEDIAFOUNDATIONMEDIAFOUNDATION_H
+#define AUDIOSOURCEMEDIAFOUNDATIONMEDIAFOUNDATION_H
+
+#include "sources/audiosource.h"
+#include "util/defs.h"
+
+#include <windows.h>
+
+#ifdef Q_OS_WIN
+#define MY_EXPORT __declspec(dllexport)
+#else
+#define MY_EXPORT
+#endif
+
+class IMFSourceReader;
+class IMFMediaType;
+class IMFMediaSource;
+
+namespace Mixxx {
+
+class AudioSourceMediaFoundation : public AudioSource {
+    typedef AudioSource Super;
+
+public:
+    static AudioSourcePointer open(QString fileName);
+
+    ~AudioSourceMediaFoundation();
+
+    diff_type seekFrame(diff_type frameIndex) /*override*/;
+
+    size_type readFrameSamplesInterleaved(size_type frameCount, sample_type* sampleBuffer) /*override*/;
+
+    void close() throw() /* override*/;
+
+private:
+    AudioSourceMediaFoundation();
+
+    Result postConstruct(QString fileName);
+
+    bool configureAudioStream();
+
+    void copyFrames(sample_type *dest, size_t *destFrames, const sample_type *src,
+            size_t srcFrames);
+
+    HRESULT m_hrCoInitialize;
+    HRESULT m_hrMFStartup;
+    IMFSourceReader *m_pReader;
+    IMFMediaType *m_pAudioType;
+    wchar_t *m_wcFilename;
+    int m_nextFrame;
```

Comment:
This limits the track length to 12:25:39.24  @ 96 kH. It seams to be not reachable. But the MediaFoundation uses a 64 bit int which allows quasi unlimited 2,000,000,000 days. So we can either optimize this code for 32 bit and deal with overflows or just use 64 bit for everything.